src/test/java/com/checkmarx/flow/cucumber/component/webhook/WebHookSteps.java

```diff
@@ -0,0 +1,106 @@
+package com.checkmarx.flow.cucumber.component.webhook;
+
+import com.checkmarx.flow.CxFlowApplication;
+import com.checkmarx.flow.config.GitHubProperties;
+import cucumber.api.PendingException;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.time.StopWatch;
+import org.awaitility.Awaitility;
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class WebHookSteps {
+    private final static Logger logger = LoggerFactory.getLogger(WebHookSteps.class);
+    private final List<CompletableFuture<Long>> requestSendingTasks = new ArrayList<>();
+    private final GitHubProperties properties;
+
+    public WebHookSteps(GitHubProperties properties) {
+        this.properties = properties;
+    }
+
+    @Given("CxFlow is running as a service")
+    public void runAsService() {
+        SpringApplication.run(CxFlowApplication.class, "--web");
+    }
+
+    @When("GitHub sends WebHook requests to CxFlow {int} times per second")
+    public void githubSendsWebHookRequests(int timesPerSecond) {
+        final int MAX_TOTAL_REQUESTS = 10,
+                MILLISECONDS_IN_SECOND = 1000;
+
+        int intervalBetweenRequestsMs = MILLISECONDS_IN_SECOND / timesPerSecond;
+
+        logger.info("Starting to send WebHook requests with the interval of {} ms.", intervalBetweenRequestsMs);
+        for (int i = 0; i < MAX_TOTAL_REQUESTS; i++) {
```

Comment:
there are better implementations for doing things in time intervals than a for loop.

Reply:
I could use Timer or ScheduledExecutorService, but they don't support specifying the number of executions, so I'll need to implement some kind of a shared variable in addition to Futures, which will complicate the code too much.